### PR TITLE
Add APIv4 Batch.create spec

### PR DESCRIPTION
**Pull request by monishdeb**
Overview
----------------------------------------
Add APIv4 Batch.create spec

Before
----------------------------------------
No default spec for APIv4 Batch.create spec

After
----------------------------------------
Add APIv4 Batch.create spec as declared [here](https://github.com/civicrm/civicrm-core/blob/master/api/v3/Batch.php#L39L53) 

Comments
----------------------------------------
ping @colemanw @eileenmcnaughton @seamuslee001 

I think this was missed in https://github.com/civicrm/civicrm-core/pull/19931 !!!

### CODE REVIEW

File: Civi/Api4/Service/Spec/Provider/BatchCreationSpecProvider.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  /**
+   * @inheritDoc
+   */
+  public function modifySpec(RequestSpec $spec) {
+    $spec->getFieldByName('created_id')->setDefaultValue('user_contact_id');
+    $spec->getFieldByName('created_date')->setDefaultValue('now');
+    $spec->getFieldByName('modified_id')->setDefaultValue('user_contact_id');
+    $spec->getFieldByName('modified_date')->setDefaultValue('now');
+    $spec->getFieldByName('status_id')->setRequired(TRUE);
```
Comment (seamuslee001) [inline Civi/Api4/Service/Spec/Provider/BatchCreationSpecProvider.php:35]: is there a reason why we aren't setting these to be required at the DB level?

Comment (monishdeb) [reply Civi/Api4/Service/Spec/Provider/BatchCreationSpecProvider.php:35]: Yeah I found that it is set true in DB level as per schema [here](https://github.com/civicrm/civicrm-core/blob/master/xml/schema/Batch/Batch.xml#L141) but in addition I think the API4 spec still need a declaration to indicate the field is required for validation and on API explorer UI as there are other such occurrences in core
```
civicrm-core$ grep -irn "setRequired(TRUE" Civi/Api4/Service/Spec/ | wc -l
      23
```

Comment (seamuslee001) [reply Civi/Api4/Service/Spec/Provider/BatchCreationSpecProvider.php:35]: I would have thought that its only needed if the db isn't requiring it right @colemanw ?

Comment (monishdeb) [reply Civi/Api4/Service/Spec/Provider/BatchCreationSpecProvider.php]: My bad, yes you are right. On further check it seems like the spec wrapper add the required property as per DAO::fields information. Updated the PR and removed the setRequired on status_id and title. Although I will submit a separate PR to remove the `setRequired` call on required (not pseudo) fields in other EntitySpecProvider files. Thanks for raising this.

File: Civi/Api4/Service/Spec/Provider/BatchCreationSpecProvider.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+   */
+  public function modifySpec(RequestSpec $spec) {
+    $spec->getFieldByName('created_id')->setDefaultValue('user_contact_id');
+    $spec->getFieldByName('created_date')->setDefaultValue('now');
+    $spec->getFieldByName('modified_id')->setDefaultValue('user_contact_id');
+    $spec->getFieldByName('modified_date')->setDefaultValue('now');
+
+    $entity_table = new FieldSpec('entity_table', 'Batch', 'String');
+    $entity_table->setTitle('Batch Entity Table - remove?');
+    $spec->addFieldSpec($entity_table);
```
Comment (colemanw) [inline Civi/Api4/Service/Spec/Provider/BatchCreationSpecProvider.php:38]: I don't understand what this extra field is for. I don't see any handling for an `entity_table` param in the `CRM_Batch_BAO_Batch::create()` function.

**Comment by monishdeb** [comment]
@eileenmcnaughton @colemanw on separate topic, is there a reason why essential field is not required in DB but required as per spec? Like civicrm_email.email is not required in DB but set required in spec : 
```
Civi/Api4//Service/Spec/Provider/EmailCreationSpecProvider.php:30:    $spec->getFieldByName('email')->setRequired(TRUE);
```
similarly civicrm_group.title, civicrm_phone.phone etc.

**Comment by colemanw** [comment]
I can't think of a good reason to not require those in the DB. It would be a small hassle to write the upgrade script so I guess no one has volunteered to do it.

**Comment by colemanw** [comment]
This PR looks complete.

**Pull request merged**